### ISSUE TITLE
[BUG]  WideAndDeepModel func should check if wide_preprocess is not None when wide_input_block is not provided

### ISSUE THREAD

**Issue by rnyak** [NVIDIA-Merlin/models#725]
### Bug description

Currently, if `wide_input_block` is not provided, it is set to None by default and in that case it can be created via `  TabularBlock.from_schema(schema=wide_schema, pre=wide_preprocess),` in these lines   [WideAndDeepModel](https://github.com/NVIDIA-Merlin/models/blob/main/merlin/models/tf/models/ranking.py#L409-L415) func. However, if `wide_preprocess` is also not provided, it is set to None, in that case no transformation block will be applied and this will result in weighted summing encoded categorical features without applying any transformation (e.g. HashedCross, etc.)

To fix this we should another check in the if statement below. something like:

```
    if not wide_input_block:
        if wide_schema is not None and len(wide_schema) > 0 and wide_preprocess is not None:
            wide_input_block = ParallelBlock(
                TabularBlock.from_schema(schema=wide_schema, pre=wide_preprocess),
                is_input=True,
                aggregation="concat",
            )
```

### Environment details
- Merlin version: 
- Platform:
- Python version:
- PyTorch version (GPU?):
- Tensorflow version (GPU?):

**Comment by rnyak** [comment]
will be fixed in https://github.com/NVIDIA-Merlin/models/pull/716

**Comment by gabrielspmoreira** [comment]
This fix was implemented differently in this #716 , as if `wide_preprocess` is not provided the `CategoryEncoding` is used for one-hot encoding.
